src/core/MOM_barotropic.F90

```diff
@@ -4163,31 +4174,27 @@ subroutine find_face_areas(Datu, Datv, G, GV, US, CS, MS, eta, halo, add_max)
 !$OMP do
     do j=js-hs,je+hs ; do I=is-1-hs,ie+hs
       Datu(I,j) = CS%dy_Cu(I,j) * GV%Z_to_H * &
-                 (max(CS%bathyT(i+1,j), CS%bathyT(i,j)) + add_max)
+                 max(max(CS%bathyT(i+1,j), CS%bathyT(i,j)) + (G%Z_ref + add_max), 0.0)
```

Comment:
Related to `Dat[uv]`, these two lines below also need to be changed. They can be problematic if both cells are active land points (negative depth) or they cancel each other out (e.g. bathyT(i,j) = 1 and bathyT(i+1,j) = -1).

https://github.com/NOAA-GFDL/MOM6/blob/060c41290bcb4a3312bdc2bab419fb59dd473c3f/src/core/MOM_barotropic.F90#L4942 

https://github.com/NOAA-GFDL/MOM6/blob/060c41290bcb4a3312bdc2bab419fb59dd473c3f/src/core/MOM_barotropic.F90#L4949

Reply:
Good catch, @herrwang0 .  These lines did not need to change to incorporate G%Z_ref because the Mean_SL already included adjustments to the reference height, so they were overlooked.  Do you have a specific change that you think would work and have already tested to avoid divisions by zero or effectively negative drag?
  If not, what would you think of changing the denominator of the first line to something like `... / ((max(G%bathyT(i+1,j) + Mean_SL, 0.0) + max(G%bathT(i,j) + Mean_SL, 0.0)) + GV%H_to_Z*GV%H_subroundoff)` ?  This would at leas avoid a division by zero, but it would give very strong drag everywhere above the mean sea-level.
  This particular block of code occurs at initialization (so the instantaneous local sea level is not known) and is inside of a test for `if (.not.CS%nonlin_stress)`. Perhaps the biggest takeaway from inspecting this line is that the linearized bottom stress is simply incompatible with wetting and drying.
  Because this line is simply reproducing the previous behavior, I think that it need not be addressed in this particular PR, but it should be addressed thoughtfully in a future PR.  If you have such a suggestion to make, @herrwang0, it would be very warmly welcomed!

Reply:
I don't really have a good change for this. For my own purpose, I just needed to add a `H_subroundoff` to make sure the denominator is not zero so that the model does not complain. I am doing tide runs and there is no surface mechanical forcing, so this part of `BT_force` is zero anyway.

I agree this is definitely not a good choice for general purposes. I guess it does make sense to use `nonlin_stress` for wetting and drying. Otherwise, maybe simply set `IDat[uv]` to zero at land points? This would be consistent with `Dat[uv]` in subroutine `find_face_areas` when there is no `eta` and `add_max` as input, and the term is calculated by harmonic means. The zero SSH approximation here does seem to imply no wetting over the land.